jax/_src/random.py

```diff
@@ -1491,6 +1491,108 @@ def _t(key, df, shape, dtype) -> Array:
+  dtype = dtypes.canonicalize_dtype(dtype)
+  if shape is not None:
+    shape = core.canonicalize_shape(shape)
+  return _chisquare(key, df, shape, dtype)
+
+@partial(jit, static_argnums=(2, 3), inline=True)
+def _chisquare(key, df, shape, dtype) -> Array:
+  if shape is None:
+    shape = np.shape(df)
+  else:
+    _check_shape("chisquare", shape, np.shape(df))
+  df = lax.convert_element_type(df, dtype)
+  two = _lax_const(df, 2)
+  half_df = lax.div(df, two)
+  log_g = loggamma(key, a=half_df, shape=shape, dtype=dtype)
+  chi2 = lax.mul(jnp.exp(log_g), two)
+  return chi2
+
+
+def f(key: KeyArray,
+      dfnum: RealArray,
+      dfden: RealArray,
+      shape: Optional[Shape] = None,
+      dtype: DTypeLikeFloat = dtypes.float_) -> Array:
+  """Sample F-distribution random values with given shape and float dtype.
+
+  Args:
+    key: a PRNG key used as the random key.
+    dfnum: a float or array of floats broadcast-compatible with ``shape``
+      representing the numerator's ``df`` of the distribution.
+    dfden: a float or array of floats broadcast-compatible with ``shape``
+      representing the denominator's ``df`` of the distribution.
+    shape: optional, a tuple of nonnegative integers specifying the result
+      shape. Must be broadcast-compatible with ``dfnum`` and ``dfden``.
+      The default (None) produces a result shape equal to ``dfnum.shape``,
+      and ``dfden.shape``.
+    dtype: optional, a float dtype for the returned values (default float64 if
+      jax_enable_x64 is true, otherwise float32).
+
+  Returns:
+    A random array with the specified dtype and with shape given by ``shape`` if
+    ``shape`` is not None, or else by ``df.shape``.
+  """
+  key, _ = _check_prng_key(key)
+  if not dtypes.issubdtype(dtype, np.floating):
+    raise ValueError("dtype argument to `f` must be a float "
+                     f"dtype, got {dtype}")
+  dtype = dtypes.canonicalize_dtype(dtype)
+  if shape is not None:
+    shape = core.canonicalize_shape(shape)
+  return _f(key, dfnum, dfden, shape, dtype)
+
+@partial(jit, static_argnums=(3, 4), inline=True)
+def _f(key, dfnum, dfden, shape, dtype) -> Array:
+  params_shapes = lax.broadcast_shapes(np.shape(dfnum), np.shape(dfden))
+  # broadcast dfnum and dfden to generate chi2 rvs
+  if not np.shape(dfnum) == params_shapes:
+      dfnum = lax.broadcast(dfnum, params_shapes)
+  if not np.shape(dfden) == params_shapes:
+      dfden = lax.broadcast(dfden, params_shapes)
+  if shape is None:
+    shape = params_shapes
+  else:
+    _check_shape("f", shape, params_shapes)
+  dfden = lax.convert_element_type(dfden, dtype)
+  dfnum = lax.convert_element_type(dfnum, dtype)
+  key_dfd, key_dfn = _split(key)
+  chi2_dfn = chisquare(key_dfn, dfnum, shape, dtype)
+  chi2_dfd = chisquare(key_dfd, dfden, shape, dtype)
+  # broadcast dfden and dfnum again to do div operation
+  dfden = lax.broadcast(dfden, shape[:-1])
+  dfnum = lax.broadcast(dfnum, shape[:-1])
+  num = lax.div(chi2_dfn, dfnum)
```

Comment:
Have you tested this with non-1D shapes? It looks like if shape is something like `(5, 6)`, this will lead to a broadcasting error, because `chi2dfn` and `dfnum` are different shapes

Reply:
Hi @jakevdp I think I should misunderstand `lax.broadcast` and `jnp.broadcast_to` , thanks for your careful review, so now I would use `jnp.broadcast_arrays` and `jnp.broadcast_to` for this shape broadcasting

Reply:
I think this time shape results are compatible with numpy